packages/artifact/src/internal/upload/upload-artifact.ts

```diff
@@ -67,34 +67,38 @@ export async function uploadArtifact(
       'CreateArtifact: response from backend was not ok'
     )
   }
-
+  // Create the zipupload stream for use in blob upload
   const zipUploadStream = await createZipUploadStream(
     zipSpecification,
     options?.compressionLevel
-  )
+  ).catch(err => {
+    throw new InvalidResponseError(
+      `createZipUploadStream: response from backend was not ok: ${err}`
+    )
+  })
```

Comment:
these awaited async functions should bubble up thrown errors, we shouldn't need this if we have an outer try/catch